Yona/Yona/Friends/FriendsDashboardViewController.swift

```diff
@@ -71,23 +71,30 @@ class FriendsDashboardViewController: MeDashBoardMainViewController {
     }
     
     override func configurProfileBarItem () {
-        if let name = buddyToShow?.UserRequestfirstName {
-            navigationItem.title = name
-            if name.count > 0 {//&& user?.characters.count > 0{
-                let btnName = UIButton()
-                let txt = "\(name.capitalized.first!)"
-                btnName.setTitle(txt, for: UIControl.State())
-                btnName.frame = CGRect(x: 0, y: 0, width: 32, height: 32)
-                btnName.addTarget(self, action: #selector(self.didChooseUserProfile(_:)), for: .touchUpInside)
+        if let firstName = buddyToShow?.UserRequestfirstName, let lastName = buddyToShow?.UserRequestlastName {
+            self.navigationItem.title = firstName + " " + lastName
+        }
+        let btnName = UIButton.init(frame: CGRect(x:0, y:0, width:YonaConstants.profileImageWidth, height:YonaConstants.profileImageHeight))
+        let rightBarButton = UIBarButtonItem()
+        if let link = buddyToShow?.buddyAvatarURL, let URL = URL(string: link) {
+            self.avtarImg.kf.setImage(with: URL, placeholder: nil, options: nil, progressBlock: nil, completionHandler: { (image, error, cacheType, imageURL) -> () in
+                let resizedImage:UIImage = UIImage.resizeImage(image: image!, targetSize: CGSize(width:YonaConstants.profileImageWidth, height:YonaConstants.profileImageHeight))
+                btnName.setImage(resizedImage, for: .normal)
                 btnName.backgroundColor = UIColor.clear
-                btnName.layer.cornerRadius = btnName.frame.size.width/2
-                btnName.layer.borderWidth = 1
-                btnName.layer.borderColor = UIColor.white.cgColor
-                let rightBarButton = UIBarButtonItem()
-                rightBarButton.customView = btnName
-                navigationItem.rightBarButtonItems = [rightBarButton]
+                btnName.clipsToBounds = true
+            })
+        } else if let nickName = buddyToShow?.buddyNickName {
+            if nickName.count > 0 {
```

Comment:
To the best of my knowledge, the nickname of a buddy is always set. If you agree there are no realistic use cases where it is missing or empty, the remove the conditions.

Reply:
I agree, there are no such cases where nick name is empty. Removed one extra if check.